featuretools/selection/selection.py

```diff
@@ -109,11 +109,14 @@ def remove_highly_correlated_features(feature_matrix, features=None, pct_corr_th
                 the feature list will not be returned. For consistent results,
                 do not change the order of features outputted by dfs.
     """
+    if feature_matrix.ww.schema is None:
+        feature_matrix.ww.init()
```

Comment:
Is the intent to remove this after the feature matrix is returned with WW initialized, or do we want this to work regardless?

If we leave it, is it possible that we would end up inferring column types to be different than they would normally be based on the primitive return type? If so, there would be no mechanism for a user to fix, and maybe it doesn't matter...

If we intend to remove, we might want to create an issue to fix this later so we don't forget.

Reply:
I think the idea is that a user might lose the woodwork typing info on their feature matrix at some point (maybe they copied it), and we want to allow them to run feature selection even in that case. @rwedge came up with the idea, I think.

Plus, the Woodwork types that get selected are `['numeric', Boolean, BooleanNullable]`, which aren't likely to have been set by us in cfm and then get reinferred incorrectly (even `Age` columns, which wouldn't get reinferred as `Age` would still get inferred as a numeric column).

It could be a problem, though, if a user has done something to their feature matrix that changes the dtypes, or if, in returning a feature matrix with Woodwork initialized slightly differently than on the original feature matrix, we'll confuse users. Maybe if woodwork wasn't initialized on the fm passed in, we un-init woodwork before returning (not sure that this is possible without a copy operation). 

I do think it makes sense to add a note to the docstring that woodwork will be initialized on the fm passed in iff need be, so if the types differ slightly, that would be why.

Reply:
@gsheni @jeff-hernandez does this behavior make sense to you?

Reply:
I think the features matrix will commonly result in numeric, categorical, or encoded categorical types. If a feature is an encoded categorical type and Woodwork sets as Categorical in the feature matrix. Would inferring without typing information change encoded categorical types to numeric?

Reply:
We should make our feature select work with OR without Woodwork typing. If the user hasn't typed their dataframe according to Woodwork (`init`), then our feature selection should do it for them. 

So we should keep the `init` in selection.py